### PR TITLE
Add methods to the SignStateMock

### PR DESCRIPTION
**Pull request by Thorinwasher**
# Description
Implemented Sign#setColor, Sign#getColor, Sign#setGlowingText and Sign#getGlowingText

# Checklist
The following items should be checked before the pull request can be merged.
- [x] Unit tests added.
- [x] Code follows existing code format.

# Info on creating a pull request
- Make sure that unit tests are added which test the relevant changes.
- Make sure that the changes follow the existing code format.

# For maintainer
When a PR is approved, the maintainer should label this PR with `release/*`.

- If the PR fixes a bug in MockBukkit, update the patch version. (if the current version is `0.4.1`, the new version should be `0.4.2`)
- If the PR adds a new feature to MockBukkit, update minor version. (if the current version is `0.4.1`, the new version should be `0.5.0`)

Note that a PR that fixes an `UnimplementedOperationException` should be considered a new version and not a bugfix.

### CODE REVIEW

File: src/main/java/be/seeseemelk/mockbukkit/block/state/SignMock.java [hunk 1]
```diff
@@ -137,29 +141,26 @@ public void setEditable(boolean editable)
...
 	@Override
 	public DyeColor getColor()
 	{
-		// TODO Auto-generated method stub
-		throw new UnimplementedOperationException();
+		return color;
 	}
 
 	@Override
 	public void setColor(DyeColor color)
```
Comment (Insprill) [inline src/main/java/be/seeseemelk/mockbukkit/block/state/SignMock.java]: This should be annotated `@NotNull` and have a Preconditions check guarding it. A test will also need to be added. In Paper it has undefined nullability, but will throw an NPE if null (`CraftSign:99` as of `20503be`)

File: src/main/java/be/seeseemelk/mockbukkit/block/state/SignMock.java [hunk 2]
```diff
@@ -137,29 +141,26 @@ public void setEditable(boolean editable)
...
-		throw new UnimplementedOperationException();
+		this.glowing = glowing;
 	}
 
 	@Override
 	public DyeColor getColor()
 	{
-		// TODO Auto-generated method stub
-		throw new UnimplementedOperationException();
+		return color;
```
Comment (Insprill) [inline src/main/java/be/seeseemelk/mockbukkit/block/state/SignMock.java]:
```suggestion
		return this.color;

File: src/main/java/be/seeseemelk/mockbukkit/block/state/SignMock.java [hunk 3]
```diff
@@ -137,29 +141,26 @@ public void setEditable(boolean editable)
 	@Override
 	public boolean isGlowingText()
 	{
-		// TODO Auto-generated method stub
-		throw new UnimplementedOperationException();
+		return glowing;
```
Comment (Insprill) [inline src/main/java/be/seeseemelk/mockbukkit/block/state/SignMock.java]:
```suggestion
		return this.glowing;

File: src/main/java/be/seeseemelk/mockbukkit/block/state/SignMock.java [hunk 4]
```diff
@@ -62,6 +64,8 @@ protected SignMock(@NotNull SignMock state)
 		{
 			lines[i] = state.getLine(i);
 		}
+		color = state.getColor();
+		glowing = state.isGlowingText();
```
Comment (Insprill) [inline src/main/java/be/seeseemelk/mockbukkit/block/state/SignMock.java]:
```suggestion
		this.color = state.getColor();
		this.glowing = state.isGlowingText();

**Review by Insprill — CHANGES_REQUESTED** [review]
Would it also be possible to implement `#setEditable`/`#isEditable`? It's also a simple getter/setting defaulting to true.

**Comment by Thorinwasher** [comment]
About also setting setEditable / getEditable. I don't know if it has something more that needs to be implemented that deals with that behavior, for example when a player tries to edit a sign block it (I really don't understand when the editable feature will be used). I'm basically not knowledgable enough to do any pull requests about that

**Comment by Insprill** [comment]
It looks like it's only used when a client sends the [update sign](https://wiki.vg/Protocol#Update_Sign) packet which we don't have implemented. If you don't want to do it that's fine, it can always be implemented later.

File: src/main/java/be/seeseemelk/mockbukkit/block/state/SignMock.java (hunk 1 above)
Comment (Thorinwasher) [reply src/main/java/be/seeseemelk/mockbukkit/block/state/SignMock.java]: I don't think I understand this. I am checking if not null in the setter, and I'm also testing it in SignMockTest#setGetColor

Comment (Insprill) [reply src/main/java/be/seeseemelk/mockbukkit/block/state/SignMock.java]: Ahh, I completely glanced over that :joy:, sorry. It just needs the NotNull annotation then

**Review by Insprill — APPROVED** [review]
LGTM. Thank you!

**Pull request merged**